spring-cloud-deployer-resource-support/src/main/java/org/springframework/cloud/deployer/resource/support/DelegatingResourceLoader.java

```diff
@@ -137,12 +137,10 @@ public Resource getResource(String location) {
 				return new FileSystemResource(cachedResource);
 			}
 		}
-		catch (URISyntaxException e) {
```

Comment:
I kept that part of your PR but somehow did not push it in my PR :(.

In my mind the general issue is to have an exception that is specifically for when this method fails, no matter the reason, otherwise we are back to the contract just being totally wide open.  

I'll push an updated PR with the changes you had in it.. I must have screwed up something there.  I'll also do the rename to ResourceNotResolved.